src/Functions/getHTTPHeader.cpp

```diff
@@ -0,0 +1,103 @@
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <DataTypes/DataTypeString.h>
+#include <Columns/ColumnString.h>
+#include <Interpreters/Context.h>
+#include <Common/CurrentThread.h>
+#include "Disks/DiskType.h"
+#include "Interpreters/Context_fwd.h"
+#include <Core/Field.h>
+#include <Poco/Net/NameValueCollection.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+}
+
+namespace
+{
+
+/** Get the value of parameter in http headers.
+  * If there no such parameter or the method of request is not
+  * http, the function will return empty string.
+  */
+class FunctionGetHTTPHeader : public IFunction, WithContext
+{
+private:
+
+public:
+    explicit FunctionGetHTTPHeader(ContextPtr context_): WithContext(context_) {}
+
+    static constexpr auto name = "getHTTPHeader";
```

Comment:
should be renamed to `getClientHTTPHeader`

Reply:
You mean change the function name to `getClientHTTPHeader` ?

Reply:
yes